relayer/crates/starknet-chain-components/src/types/messages/ibc/denom.rs

```diff
@@ -195,14 +199,14 @@ mod tests {
         let input = "transfer2/channel-1/transfer/channel-0/coin";
         let expected = PrefixedDenom {
             trace_path: vec![
-                TracePrefix {
-                    port_id: "transfer2".to_string(),
-                    channel_id: "channel-1".to_string(),
-                },
                 TracePrefix {
                     port_id: "transfer".to_string(),
                     channel_id: "channel-0".to_string(),
                 },
+                TracePrefix {
+                    port_id: "transfer2".to_string(),
+                    channel_id: "channel-1".to_string(),
+                },
```

Comment:
this reminds me, I am computing the token hash here incorrectly.

https://github.com/informalsystems/ibc-starknet/blob/a1028feac57ff73b94667d90b782a418186e027b/relayer/crates/starknet-integration-tests/src/tests/ics20.rs#L480-L482

this should be

```rs
 for trace_prefix in ibc_prefixed_denom.trace_path.iter().rev() { 
```

You copied my code in `clear_packet.rs` too. So, the same is present there too.

Reply:
You mean in `from_cosmos_to_cairo_packet`, https://github.com/informalsystems/ibc-starknet/blob/main/relayer/crates/starknet-chain-components/src/impls/messages/packet.rs#L293-L304?